Assembler.Base/InitialMessageHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class InitialMessageHandler<TFrame, TMessage> : BaseMessageHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        private readonly ILogger _logger;
+
+        public InitialMessageHandler(ITimeBasedCache<TMessage> cache,
+            IFactory<TFrame, string> identifierFactory, ICreator<TMessage> assembledMessageCreator,
+            IMessageEnricher<TFrame, TMessage> enricher, ILoggerFactory loggerFactory) : base(cache,
+            identifierFactory, enricher, assembledMessageCreator)
+        {
+            _logger = loggerFactory.GetLogger(this);
+        }
+
+        public override void Handle(TFrame frame)
```

Comment:
For all 3 non-abstract `MessageHandler` classes:
1. The entire ID-creation process is duplicated in all 3 classes, I heavily recommend moving it to the base class.
2. All 3 implementations of the `Handle(TFrame frame)` method result in **VERY** big chunks of code. It's not unreadable, but it's a bit of an eyesore. I think you should try splitting these up into several methods, just so you won't have 50-lines-long methods.

Reply:
1. I know, the reason I'm doing it is because I want the logger to be correlated with the class. 
2. You are right. Fixed.

Reply:
1. A suggestion, if you wish to accept it - create a `TryGetIdentifer` method in the base-class, and **accept `ILogger` as a parameter**. That way you might have the best of both worlds (although having `ILogger` as a parameter might makes some people go "Hmmm...")
2. Neat 👍

Reply:
Check it now